components/server/src/session-handler.ts

```diff
@@ -40,7 +42,25 @@ export class SessionHandlerProvider {
 
         options.store = this.createStore();
 
-        this.sessionHandler = session(options);
+        this.sessionHandler = async (req: express.Request, res: express.Response) => {
```

Comment:
I'm not sure this will work: We should double-check that whoever calls that handler can actually handle `async`. This is typically the case with express (it just ignores the return values), but: Did you test this?

Also, we should have a "try-catch" around the feature flag retrieval, to make sure we don't block session creation.

Reply:
> Also, we should have a "try-catch" around the feature flag retrieval, to make sure we don't block session creation.

It's handled inside the retrieval, if it fails, it serves the default.

Reply:
No need to add now: But this stuff would be great to have a comment for (in other cases, in the future).

Reply:
Sure, but then we'd have to do this for every invocation of `getValueAsync()` which also doesn't feel appropriate. Possibly renaming the `getValueAsync` to something which indicates that would be better.